[clang] Robustify openmp test

urnathan opened the pull request:
If the source path contains 'alias' this would spuriously fail.  Be more specific about not wanting [no]alias annotations.

urnathan committed:
[clang] Robustify open mp test

If the source path contains 'alias' this would spuriously fail.  Be
more specific about not wanting [no]alias annotations.

jdoerfert reviewed (commented):
Your motivation is sound, apologies for the churn this caused.

Thaat said, this alias is not `noalias` but global symbols aliases.
The commit that introduced the NOT check removed:
```
// CHECK-DAG: @_ZN16SpecSpecialFuncs6MethodEv = {{.*}}alias i32 (%struct.SpecSpecialFuncs*), i32 (%struct.SpecSpecialFuncs*)* @_ZN16SpecSpecialFuncs7method_Ev
// CHECK-DAG: @_ZN16SpecSpecialFuncs6methodEv = linkonce_odr {{.*}}alias i32 (%struct.SpecSpecialFuncs*), i32 (%struct.SpecSpecialFuncs*)* @_ZN16SpecSpecialFuncs7method_Ev
```
So we want to avoid `@ ... = ... alias` or similar.

urnathan committed:
[clang] Robustify open mp test

If the source path contains 'alias' this would spuriously fail.  Be
more specific about not wanting global aliases, which is what
introduced this check.

urnathan commented:
@jdoerfert thanks for that, here's an update that focusses the check-not more precisely.

jdoerfert reviewed (approved):
LG, thx
